staging/src/k8s.io/apiserver/pkg/clientsethack/adapter.go

```diff
@@ -0,0 +1,328 @@
+	eventsv1beta1 "k8s.io/client-go/kubernetes/typed/events/v1beta1"
+	extensionsv1beta1 "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
+	flowcontrolv1 "k8s.io/client-go/kubernetes/typed/flowcontrol/v1"
+	flowcontrolv1beta1 "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta1"
+	flowcontrolv1beta2 "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta2"
+	flowcontrolv1beta3 "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta3"
+	networkingv1 "k8s.io/client-go/kubernetes/typed/networking/v1"
+	networkingv1alpha1 "k8s.io/client-go/kubernetes/typed/networking/v1alpha1"
+	networkingv1beta1 "k8s.io/client-go/kubernetes/typed/networking/v1beta1"
+	nodev1 "k8s.io/client-go/kubernetes/typed/node/v1"
+	nodev1alpha1 "k8s.io/client-go/kubernetes/typed/node/v1alpha1"
+	nodev1beta1 "k8s.io/client-go/kubernetes/typed/node/v1beta1"
+	policyv1 "k8s.io/client-go/kubernetes/typed/policy/v1"
+	policyv1beta1 "k8s.io/client-go/kubernetes/typed/policy/v1beta1"
+	rbacv1 "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	rbacv1alpha1 "k8s.io/client-go/kubernetes/typed/rbac/v1alpha1"
+	rbacv1beta1 "k8s.io/client-go/kubernetes/typed/rbac/v1beta1"
+	resourcev1alpha3 "k8s.io/client-go/kubernetes/typed/resource/v1alpha3"
+	resourcev1beta1 "k8s.io/client-go/kubernetes/typed/resource/v1beta1"
+	schedulingv1 "k8s.io/client-go/kubernetes/typed/scheduling/v1"
+	schedulingv1alpha1 "k8s.io/client-go/kubernetes/typed/scheduling/v1alpha1"
+	schedulingv1beta1 "k8s.io/client-go/kubernetes/typed/scheduling/v1beta1"
+	storagev1 "k8s.io/client-go/kubernetes/typed/storage/v1"
+	storagev1alpha1 "k8s.io/client-go/kubernetes/typed/storage/v1alpha1"
+	storagev1beta1 "k8s.io/client-go/kubernetes/typed/storage/v1beta1"
+	storagemigrationv1alpha1 "k8s.io/client-go/kubernetes/typed/storagemigration/v1alpha1"
+)
+
+// Interface allows us to hold onto a strongly-typed cluster-aware clients here, while
+// passing in a cluster-unaware (but non-functional) clients to k8s libraries. We export this type so that we
+// can get the cluster-aware clients back using casting in admission plugin initialization.
+type Interface interface {
+	kubernetes.Interface
+	ClusterAware() kcpkubernetesclientset.ClusterInterface
+}
+
+var _ Interface = (*hack)(nil)
+
+// Wrap adapts a cluster-aware informer factory to a cluster-unaware wrapper that can divulge it after casting.
+func Wrap(clusterAware kcpkubernetesclientset.ClusterInterface) Interface {
+	return &hack{clusterAware: clusterAware}
+}
+
+// Unwrap extracts a cluster-aware informer factory from the cluster-unaware wrapper, or panics if we get the wrong input.
+func Unwrap(clusterUnaware kubernetes.Interface) kcpkubernetesclientset.ClusterInterface {
+	return clusterUnaware.(Interface).ClusterAware()
+}
+
+type hack struct {
+	clusterAware kcpkubernetesclientset.ClusterInterface
+}
+
+func (h *hack) AdmissionregistrationV1alpha1() admissionregistrationv1alpha1.AdmissionregistrationV1alpha1Interface {
```

Comment:
im surprised this didn't changed

Reply:
The `Interface` interface seems to still have that method in upstream's v1.32: https://github.com/kubernetes/kubernetes/blob/v1.32.0/staging/src/k8s.io/client-go/kubernetes/clientset.go#L27

Unless you meant something else, @mjudeikis ?

Reply:
I mean no new types were added/removed in 1.32.

Reply:
`git diff kcp-1.31.0..kcp-1.32-pre.7-review -- staging/src/k8s.io/apiserver/pkg/clientsethack/adapter.go` says that:
* `coordinationV1alpha1` was replaced by `coordinationV1alpha2`, and
* `resourcev1beta1` was added

Should I document this somewhere?